### PR TITLE
[vcpkg] Add `versions` feature flag and version field manifest parsing

### PR DESCRIPTION
**Pull request by ras0219**
* Introduce FeatureFlagSettings struct to more easily access feature flags throughout the program
* To avoid users accidentally starting to write "version" instead of "version-string" in their manifests, vcpkg explicitly detects and prevents usage of ports with schemes other than "String"

Example output when changing `abseil`'s field to just `"version"`:
```
PS C:\src\vcpkg> ./vcpkg install abseil
Computing installation plan...
Error: while loading port `abseil`: Port definition rejected because the `versions` feature flag is disabled.
This can be fixed by using the "version-string" field.
```

* Drive-by fix of copiable SourceControlFileLocation and an exposed use-after-move bug

This code is largely extracted from PR #13777

### CODE REVIEW

File: toolsrc/include/vcpkg/versions.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <vcpkg/fwd/vcpkgpaths.h>
+
+#include <set>
+#include <string>
+#include <tuple>
+#include <vector>
```
Comment (vicroms) [inline toolsrc/include/vcpkg/versions.h:8]: Don't need all these `#include`

**Review by vicroms — APPROVED** [review]

**Comment by vicroms** [comment]
Are you adding versioning scheme validation in this PR? 
E.g.: `version-dates` follows the date format described in the spec, etc.

**Comment by ras0219** [comment]
I omitted it from this PR for simplicity

**Pull request merged**